### PR TITLE
Retrieval metrics

### PR DESCRIPTION
**Pull request by assaftibm**

### CODE REVIEW

File: src/unitxt/metrics.py [hunk 1]
```diff
@@ -1278,3 +1279,155 @@ def compute(
...
+    ) -> dict:
+        # digest input
+        pred_ids: List[Any] = prediction
+        ref_ids: List[Any] = list(dict.fromkeys(references))
+
+        # relevance_at_k: 1-based dictionary of indicators (0/1), telling whether
+        # the doc id retrieved at position k (assuming it is 1-based, so k starts
+        # from 1) is in the gold doc ids or not.
+        # For example, assuming that in the retrieved docs we have correct predictions
+        # at positions 2, 4 and 5 (1-based), the dict will look like like:
```
Comment (dafnapension) [inline src/unitxt/metrics.py:1296]:
```suggestion
        # at positions 2, 4 and 5 (1-based), the dict will look like:
```

File: src/unitxt/metrics.py [hunk 2]
```diff
@@ -1278,3 +1279,155 @@ def compute(
...
+        # {1: 0, 2: 1, 3: 0, 4: 1, 5: 1, ....]
```
Comment (dafnapension) [inline src/unitxt/metrics.py:1297]:
```suggestion
        # {1: 0, 2: 1, 3: 0, 4: 1, 5: 1, ...}
```

File: prepare/metrics/mrr.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+metric = MRR()
+
+predictions = [["a", "b", "c", "d", "e", "f"], ["g", "r", "u"], ["a", "b"], [], ["c"]]
+
+references = [
+    ["c", "d"],  # third hit
+    ["g"],  # first hit
+    [],  # no hit
+    ["a"],  # no hit
+    ["b"],
```
Comment (dafnapension) [inline prepare/metrics/mrr.py:14]:
```suggestion
    ["b"],   # no hit
```

File: prepare/metrics/mrr.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+]  # no hit
```
Comment (dafnapension) [inline prepare/metrics/mrr.py:15]:
```suggestion
] 
```

**Review by dafnapension — APPROVED** [review]

**Comment by dafnapension** [comment]
A pleasure to read. I left some very petty comments.

**Comment by assaftibm** [comment]
@dafnapension I'm so glad to see you here! :-) 
and thanks for the r review!

**Pull request merged**